tests/conftest.py

```diff
@@ -78,236 +74,78 @@ def openai_api_key(monkeypatch: pytest.MonkeyPatch) -> str:
-
-            content = asyncio.run(read_stream())
-            return Response(
-                status_code=response.status_code,
-                headers=response.headers,
-                content=content,
-                request=request,
-            )
-
-    class AsyncTransport(httpx.AsyncBaseTransport):
-        def __init__(self, asgi_transport: httpx.ASGITransport):
-            self.asgi_transport = asgi_transport
-
-        async def handle_async_request(self, request: Request) -> Response:
-            response = await self.asgi_transport.handle_async_request(request)
-
-            async def read_stream():
-                content = b""
-                async for chunk in response.stream:
-                    content += chunk
-                return content
-
-            content = await read_stream()
-            return Response(
-                status_code=response.status_code,
-                headers=response.headers,
-                content=content,
-                request=request,
-            )
-
-    asgi_transport = httpx.ASGITransport(app=test_app)
-    sync_transport = SyncTransport(asgi_transport=asgi_transport)
-    sync_client = httpx.Client(transport=sync_transport, base_url="http://test")
-    async_transport = AsyncTransport(asgi_transport=asgi_transport)
-    async_client = httpx.AsyncClient(transport=async_transport, base_url="http://test")
-    return sync_client, async_client
+    yield engine
+    await engine.dispose()
 
 
 @pytest.fixture
-def prod_db(request, dialect) -> async_sessionmaker:
-    """
-    Instantiates DB in a manner similar to production.
-    """
+def db(
+    request: SubRequest,
+    dialect: str,
+) -> Callable[[], AsyncContextManager[AsyncSession]]:
     if dialect == "sqlite":
-        return request.getfixturevalue("prod_sqlite_db")
+        return _db_with_lock(request.getfixturevalue("sqlite_engine"))
     elif dialect == "postgresql":
-        return request.getfixturevalue("prod_postgres_db")
+        return _db_with_lock(request.getfixturevalue("postgresql_engine"))
     raise ValueError(f"Unknown db fixture: {dialect}")
 
 
-@pytest.fixture
-def prod_sqlite_db(
-    sqlite_engine: AsyncEngine,
-) -> Callable[[], AsyncContextManager[AsyncSession]]:
-    """
-    Instantiates SQLite in a manner similar to production.
-    """
-    Session = async_sessionmaker(sqlite_engine, expire_on_commit=False)
-
-    @contextlib.asynccontextmanager
-    async def factory() -> AsyncIterator[AsyncSession]:
-        async with Session.begin() as session:
-            yield session
-
-    return factory
-
-
-@pytest.fixture
-def prod_postgres_db(
-    postgres_engine: AsyncEngine,
-) -> Callable[[], AsyncContextManager[AsyncSession]]:
-    """
-    Instantiates Postgres in a manner similar to production.
-    """
-    Session = async_sessionmaker(postgres_engine, expire_on_commit=False)
+def _db_with_lock(engine: AsyncEngine) -> Callable[[], AsyncContextManager[AsyncSession]]:
```

Comment:
I do think it might be worth only wrapping Postgres if it's only Postgres that needs this for the sake of keeping the test and prod environments as similar as possible.

Reply:
We're going to put locks on sqlite for production soon, plus adding locks would only help and not hurt since our goal is only to test SQL syntax and not concurrency.